src/thor/alternates.cc

```diff
@@ -56,7 +62,74 @@ void filter_alternates_by_stretch(std::vector<CandidateConnection>& connections)
   connections.erase(new_end, connections.end());
 }
 
-// Limited Sharing. Compare duration of edge segments shared between optimal path and
+// get a cost of a path between indexes 'first' and 'last'
+inline sif::Cost get_segment_cost(const std::vector<PathInfo>& path, size_t first, size_t last) {
+  auto cost = path[last].elapsed_cost - path[first].transition_cost;
+  if (first > 0)
+    cost -= path[first - 1].elapsed_cost;
+  return cost;
+};
+
+// Find a different segment for two routes. By design bidirectional astar returns routes that have
+// only one different segment. Also the first and last edges are the same for each two routes.
+std::pair<std::pair<size_t, size_t>, std::pair<size_t, size_t>>
+find_diff_segment(const std::vector<PathInfo>& path_1, const std::vector<PathInfo>& path_2) {
+
+  size_t idx_1_first = 0;
+  size_t idx_2_first = 0;
+
+  // find first different edge
+  while (idx_1_first < path_1.size() && idx_2_first < path_2.size() &&
+         path_1[idx_1_first].edgeid == path_2[idx_2_first].edgeid) {
+    ++idx_1_first;
+    ++idx_2_first;
+  }
+  // check corner cases: stop if we didn't find a different edge
+  if (idx_1_first == path_1.size())
+    return {{idx_1_first, idx_1_first}, {idx_2_first, std::max(idx_2_first, path_2.size() - 1)}};
+  else if (idx_2_first == path_2.size())
+    return {{idx_1_first, std::max(idx_1_first, path_1.size() - 1)}, {idx_2_first, idx_2_first}};
+
+  size_t idx_1_last = path_1.size() - 1;
+  size_t idx_2_last = path_2.size() - 1;
+  // find last different edge
+  while (idx_1_last > idx_1_first && idx_2_last > idx_2_first &&
+         path_1[idx_1_last].edgeid == path_2[idx_2_last].edgeid) {
+    --idx_1_last;
+    --idx_2_last;
+  }
+
+  return {{idx_1_first, idx_1_last}, {idx_2_first, idx_2_last}};
+}
+
+// Check if the candidate path contains unreasonable long detours comparing to the optimal path.
+bool validate_alternate_by_stretch(const std::vector<PathInfo>& optimal_path,
+                                   const std::vector<PathInfo>& candidate_path) {
+  const auto diff_segment = find_diff_segment(optimal_path, candidate_path);
+  const auto& idxs_opt = diff_segment.first;
+  const auto& idxs_alt = diff_segment.second;
```

Comment:
this naming is difficult for my brain. `idxs_opt` is the segment of the optimal path which is different and `idxs_alt` is the segment of the candidate path that is different. what about like `unique_optimal_segment` and `unique_candidate_segment`? maybe also `diff_segment` would be like `unique_segments`? i dont know just a suggestion, though pretty long names...

Reply:
I like that, thank you !